src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -229,6 +233,31 @@ private function extractFromConstructor(string $class, string $property): ?array
         return null;
     }
 
+    private function extractFromDefaultValue(string $class, string $property)
+    {
+        try {
+            $reflectionClass = new \ReflectionClass($class);
+        } catch (\ReflectionException $e) {
+            return null;
+        }
+
+        $defaultValues = $reflectionClass->getDefaultProperties();
+
+        if (!isset($defaultValues[$property])) {
+            return null;
+        }
+
+        $map = array(
+            'integer' => Type::BUILTIN_TYPE_INT,
+            'boolean' => Type::BUILTIN_TYPE_BOOL,
+            'double' => Type::BUILTIN_TYPE_FLOAT,
+        );
+
+        $type = gettype($defaultValues[$property]);
```

Comment:
I think it's safer to not pass (`return null`) if the type is null.
Most of the time, it will mean that the property isn't initialized (and it may be initialized in the constructor, with null values not allowed). Even when `null` is intended, it's very unlikely that it will be the only type allowed.

Returning an array means that next extractors in the chain will be skipped, it shouldn't be the case if the value is `null` (another extractor may find the relevant type).